content/docs/atom-guide/dev-guide/shaders/azsl/_index.md

```diff
@@ -1,20 +1,468 @@
+  
+Classes has no constructors/destructors and as such, do not accept default member initializers.
+
+### Matrices
+AZSL follows the HLSL convention to use column major matrices by default.  
+  
+However, in the O3DE Runtime, the default convention is row major matrices.  
+  
+When declaring a matrix, the order can be specified using the row_major or column_major keywords.  
+  
+    ShaderResourceGroupSemantic MatrixExample
+    {
+        FrequencyId = 1;
+    };
+      
+    ShaderResourceGroup MatrixExampleSRG : MatrixExample
+    {
+                     float2x3 defaultMatrix;
+        row_major    float2x3 rowMajorMatrix;
+        column_major float2x3 colMajorMatrix;
+    };
+  
+### Arrays And Unbounded Arrays
+From the point of view of defining arrays, the rules and limitations are exactly the same as HLSL. Examples:  
+  
+    ShaderResourceGroup PerObject : BindingPerObject
+    {
+        float m_arrayOfFloats[64]; // GOOD. Arrays of fundamental types must be of constant size.
+     
+        struct SomeStruct
+        {
+            int m_integer;
+            bool m_boolean;
+        }
+        static const int MyArraySize = 32;
+        SomeStruct m_arrayOfStructs[MyArraySize]; // GOOD. Arrays of packable types (structs & fundamentals) require constant size definition.
+     
+        int m_unboundedArrayOfIntegers[]; // ERROR. Unbounded arrays are not supported for fundamental(packables, in general) data types.
+     
+        Texture2D<float4> m_textures[MyArraySize]; // GOOD. Contant size array of Resource View types is ok.
+         
+        ConstantBuffer<SomeStruct> m_unboundedArrayOfStructs[]; // GOOD. Unbounded array of Resource View type is ok. BUT, there are some restrictions.
+    }
+  
+To learn more about what are the limitations when declaring unbounded arrays, go here: [Binding Rules For Unbounded Arrays.](unbounded-arrays)  
+  
+### <a name="partial"></a> Partial ShaderResourceGroup Definitions.
+Before talking about `partial ShaderResourceGroup`, it's important to talk about [ShaderResourceGroupSemantic](srg-semantics), which is always required when declaring a `ShaderResourceGroup`.  
+Go to this section to read about [ShaderResourceGroupSemantic](srg-semantics).  
+  
+The `partial` keyword is useful to define parts of an SRG in different blocks. For example, across different files. This is a similar concept as partial classes in C#.  
+There are no limits into how many `partial` blocks can be used to define an SRG.  
+There are only three requirements:  
+1. When using `partial` to define an SRG block, all other SRG definition blocks, for the same SRG, must start with the `partial` keyword.
+2. For a given SRG, at least one `partial` SRG block must define the Shader Resource Group Semantic.
+3. For a given SRG, if the Shader Resource Group Semantic is defined more than once across different `partial` blocks, they must refer to the same `ShaderResourceGroupSemantic`.  
+  
```

Comment:
- I suggest moving the remark you wrote below here, before the examples.
  Original text:
  > An important remark about partial SRG blocks is that when the compiler, **AZSLc**, finds the first `partial` block, it will use it as the unified point of emission for all the data of a given SRG across all `partial` blocks. The main recommendation that comes out of this rule, is that any globally defined `struct`, or `class` that may be referenced by a `partial` SRG block must be defined before the first partial block for the SRG in question.  

- If it's important, you can use {{< note >}} ... {{< /note>}} to create a Note callout box. To see this, preview the docs in a website preview. 

- Specify "SRG" when talking about "partial SRG".
- Don't use in-line code style for words such as classes, structs, partials, unless you are specifically referring to the keyword. For example: "Use the `partial` keyword to create partial blocks."

```suggestion
{{< note >}}
During compilation, **AZSLc** uses the first partial SRG block it finds as the unified point of emission for all the data of a given SRG across all partial SRG blocks. Therefore, we recommend that you define structs and classes that are referenced by a partial SRG before the first partial block.
{{< /note >}}
```

Reply:
Fixed